app/scripts/metamask-controller.js

```diff
@@ -6110,23 +6295,16 @@ export default class MetamaskController extends EventEmitter {
       outStream,
       (err) => {
         // handle any middleware cleanup
-        engine._middleware.forEach((mid) => {
-          if (mid.destroy && typeof mid.destroy === 'function') {
-            mid.destroy();
-          }
-        });
+        engine.destroy();
```

Comment:
?

Reply:
the previous logic was unnecessarily iterating over the middleware directly. Calling `engine.destroy()` does the same. This also matches how we handle the underlying stream ending for the EIP-1193 API (i.e. we also call `engine.destroy()` in that pipeline)